buildSrc/src/main/groovy/com/uber/okbuck/generator/BuckFileGenerator.groovy

```diff
@@ -97,7 +94,7 @@ final class BuckFileGenerator {
                     }
                     break
                 default:
-                    break
+                    throw new IllegalArgumentException("ProjectType not handled: $projectType")
```

Comment:
This will break users who forget to set buck projects in the extension. That's a breaking API change. What's the reason for this?